QAchat/python/QAchat.ipynb

```diff
@@ -0,0 +1,428 @@
+   "metadata": {},
+   "source": [
+    "## 1. Load in datasets\n",
+    "\n",
+    "First, we'll load two datasets:\n",
+    "1. An evaluation dataset to test our pipeline. This includes input, output pairs like:\n",
+    "```\n",
+    "    {\"input\": \"What is my full name?\", \"output\": \"John Smith. -BT\"}\n",
+    "```\n",
+    "2. A user context dataset to give to the AI assistant as context. This will be stored in a vector db and contains rows like:\n",
+    "```\n",
+    "    {\"category\": \" address\", \"detail\": \"123 Main Street, Anytown, USA\"}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "# Input, output pairs to evaluate our QAChatbot.\n",
+    "# Upload these to a dataset in BrainTrust so your teammates can also easily use and manage this dataset.\n",
+    "dataset = braintrust.init_dataset(PROJECTNAME, name=\"Basic Evaluation\", api_key=BT_API_KEY)\n",
+    "with open('evaluationDataset.jsonl') as f:\n",
+    "    for line in f:\n",
+    "        testCase = json.loads(line)\n",
+    "        dataset.insert(\n",
+    "            input=testCase['input'],\n",
+    "            output=testCase['output'],\n",
+    "        )\n",
+    "\n",
+    "# List of facts about a fake user. These will be given as context to the QAChatbot.\n",
+    "userContextDataset = []\n",
+    "with open('userContextDataset.jsonl') as f:\n",
+    "    for line in f:\n",
+    "        userContextDataset.append(json.loads(line))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Store the user context dataset\n",
+    "Next, we will embed the personal dataset of information into a vector DB. We will use [Chroma](https://www.trychroma.com/) to embed, store, and retrieve the relevant user context."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "chroma_client = chromadb.Client()\n",
+    "collection = chroma_client.create_collection(name=\"user-context\")\n",
+    "\n",
+    "for i,c in enumerate(userContextDataset):\n",
+    "    fact = c[\"detail\"]\n",
+    "    collection.add(\n",
+    "        documents = [fact],\n",
+    "        ids = [str(i)]\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Write an evaluation function\n",
+    "\n",
+    "Now, we will define a general evaluation function to test different prompts and pipelines using BrainTrust. This makes it easy to iterate and improve our AI app."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "# This is an evaluation script that tests a prompt and pipeline combination and logs the results to BrainTrust.\n",
+    "def runEvaluation(generationFn, experimentName):\n",
```

Comment:
Everything should be snake-cased to be consistent with every other notebook (i.e. `run_evaluation`).